### PR TITLE
Add org.dupot.beatandmatchtopass

### PR DESCRIPTION
**Pull request by imikado**
### Please confirm your submission meets all the criteria

- [x ] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ x] My pull request follows the instructions at [App Submission][submission].
- [ x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [ x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by imikado** [comment]
I use previous littleAdventure branch to commit this new game

File: org.dupot.beatandmatchtopass.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+id: org.dupot.beatandmatchtopass
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
```
Comment (hfiguiere) [inline org.dupot.beatandmatchtopass.yml:3]: this runtime is EOL.

File: org.dupot.beatandmatchtopass.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  - --device=dri
+  - --persist=.local/share/godot/app_userdata/BeatMatchtoPass
+
+modules:
+  - name: BeatMatchtoPass
+    buildsystem: simple
+    build-commands:
+      - install -Dm555 export/linux/beatmatchtopass.x86_64 /app/bin/beatmatchtopass.x86_64
+      - install -Dm644 export/linux/flatpack/512x512.png /app/share/icons/hicolor/512x512/apps/org.dupot.beatmatchtopass.png
+      - install -Dm644 export/linux/flatpack/org.dupot.beatmatchtopass.desktop /app/share/applications/org.dupot.beatmatchtopass.desktop
```
Comment (bbhtt) [inline org.dupot.beatandmatchtopass.yml:19]: https://github.com/imikado/dupotBeatAndMatchToPass/blob/67167b59b4bef2ebe6246400471df0949146bb58/export/linux/flatpack/org.dupot.beatmatchtopass.desktop#L3 should not be absolute path

File: org.dupot.beatandmatchtopass.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      - install -Dm644 export/linux/flatpack/org.dupot.beatmatchtopass.appdata.xml /app/share/metainfo/org.dupot.beatmatchtopass.appdata.xml
```
Comment (bbhtt) [inline org.dupot.beatandmatchtopass.yml:20]: https://github.com/imikado/dupotBeatAndMatchToPass/blob/67167b59b4bef2ebe6246400471df0949146bb58/export/linux/flatpack/org.dupot.beatmatchtopass.appdata.xml#L16-L34 Pull screenshots from a tag/commit not main

File: org.dupot.beatandmatchtopass.yml (hunk 2 above)
Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: fixed

File: org.dupot.beatandmatchtopass.yml (hunk 3 above)
Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: fixed

File: org.dupot.beatandmatchtopass.yml (hunk 2 above)
Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: how?

File: org.dupot.beatandmatchtopass.yml (hunk 3 above)
Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: how?

Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: i push modifications for this file, i replace absolute by $FLATPAK_DEST/share...

Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: that's not what was asked here. You didn't address the `Pull screenshots from a tag/commit not main`

Still pull from main

image

**Comment by hfiguiere** [comment]
bot, build org.dupot.beatandmatchtopass

File: org.dupot.beatandmatchtopass.yml (hunk 2 above)
Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: still doing it 
image

File: org.dupot.beatandmatchtopass.yml (hunk 3 above)
Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: i just fixed it

Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: (i just commit)

File: org.dupot.beatandmatchtopass.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  - --persist=.local/share/godot/app_userdata/BeatMatchtoPass
+
+modules:
+  - name: BeatMatchtoPass
+    buildsystem: simple
+    build-commands:
+      - install -Dm555 export/linux/beatmatchtopass.x86_64 $FLATPAK_DEST/bin/beatmatchtopass.x86_64
+      - install -Dm644 export/linux/flatpack/512x512.png $FLATPAK_DEST/share/icons/hicolor/512x512/apps/org.dupot.beatmatchtopass.png
+      - install -Dm644 export/linux/flatpack/org.dupot.beatmatchtopass.desktop $FLATPAK_DEST/share/applications/org.dupot.beatmatchtopass.desktop
+      - install -Dm644 export/linux/flatpack/org.dupot.beatmatchtopass.appdata.xml $FLATPAK_DEST/share/metainfo/org.dupot.beatmatchtopass.appdata.xml
```
Comment (hfiguiere) [inline org.dupot.beatandmatchtopass.yml:20]: this doesn't match the app-id in the manifest.

File: org.dupot.beatandmatchtopass.yml (hunk 2 above)
Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: no. that's even worse.

Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: i just commit the fix

File: org.dupot.beatandmatchtopass.yml (hunk 4 above)
Comment (imikado) [reply org.dupot.beatandmatchtopass.yml:20]: i fix it in lasts commit

File: org.dupot.beatandmatchtopass.yml (hunk 2 above)
Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: you didn't. you did what I said to NOT do.

File: org.dupot.beatandmatchtopass.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-id: org.dupot.beatandmatchtopass
+id: org.dupot.beatmatchtopass
```
[the remark this answers is not in the document]
Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: commit fix

File: org.dupot.beatandmatchtopass.yml (hunk 2 above)
Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: i delete $FLATHPAK_DEST, only bin/..

Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: ok, i remove .x86_64 (for information, it is the default godot generated name )
just commit

Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: :facepalm:

Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: *hint* `bin` is a directory. it is not part of the executable file name. `Exec` is supposed to be the executable file name, as it is supposed to be located in one of the directories in `PATH`. This was the original request to no have an absolute path.

Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: i don't understand: this is the godot executable binary

Comment (hfiguiere) [reply org.dupot.beatandmatchtopass.yml]: `beatmatchtopass.x86_64` IS.

Comment (imikado) [reply org.dupot.beatandmatchtopass.yml]: ok
scuse me, but as mentionned, i only copied last published project and change only project name.. i don't know why this was ok for my previous projet but not this one
i commit only executable name

**Pull request closed without merging**